UnityProject/Assets/Scripts/Systems/GhostRoles/GhostRole.cs

```diff
@@ -153,6 +155,7 @@ private void EnableDefaultRespawning()
 		private void SpawnPlayer(PlayerInfo player)
 		{
 			playersSpawned++;
+			player.Mind.CurrentCharacterSettings = CharacterSheet.GenerateRandomCharacter();
```

Comment:
Maybe should be a setting on the ghost role itself, iif it Re-roles the character?

Reply:
there's no case where you want ghost roles to carry the same name currently.